tidev/hyperloop.next
# fix(android): check for inner-class and field name collision

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27915

**Summary:**
Fixes JS runtime error "Cannot redefine property" caused by Java/Kotlin class having an inner-class and a field with the same name. Will always happen when using the Kotlin "companion" feature.

**Test:**
1. Download the [hyperloop-examples](https://github.com/appcelerator/hyperloop-examples) project.
2. Add the below dependency to the project's "build.gradle" file.
3. Add the below line of JS code to the "alloy.js" file.
4. Build and run on Android.
5. Verify the app starts up without issue.

./app/platform/android/build.gradle
```groovy
dependencies {
	implementation 'com.stripe:stripe-android:14.4.1'
}
```

./app/alloy.js
```javascript
var CardInputWidget = require("com.stripe.android.view.CardInputWidget");
```
